src/Psalm/Internal/Type/TemplateInferredTypeReplacer.php

```diff
@@ -451,6 +451,12 @@ private static function replaceConditional(
                     null,
                     false,
                     false,
+                ) && null === Type::intersectUnionTypes(
```

Comment:
The logic was

When doing `A is B ? C : D`:
- If A is contained by B => It's C
- If B is not contained by A => It's D
- Else, it's C|D

I think this logic is wrong and does not work well with string type like `literal-string` and `non-empty-string`.
`literal-string is non-empty-string ? int : float`.
Since `non-empty-string` is not contained by `literal-string` it was considering the result as a float.
But since `literal-string&non-empty-string` is non empty, the result should have been `int|float`.